### PR TITLE
[WIP] PR to address JWSTSIAF 164

### PR DESCRIPTION
**Pull request by mgennaro**
We are delivering new apertures for these tickets. We are also adding placeholder distortion files for the new full-frame master apertures for coronographic modes. These distortion files will eventually account for the changes in distortion solution due to the introduction of the wedge into the optical path

@JAStansberry @marthaboyer

### CODE REVIEW

File: pysiaf/source_data/NIRCam/nircam_siaf_aperture_definition.txt [hunk 1]
```diff
@@ -49,6 +49,33 @@
 #            NRCB5_SUB400P  ## updated to match B1_400P   
 #              NRCB5_FULLP  ## updated to match B1_FULLP
 #
+#
+# gennaro 2019-12-12
+#  Added 96 WFSS filter-dependent SLIT type apertures for visualization. Apertures for both LW detectors, and for
```
Comment (shanosborne) [inline pysiaf/source_data/NIRCam/nircam_siaf_aperture_definition.txt]: Remove JWSTSIAF-67 references from this PR

File: pysiaf/source_data/NIRCam/nircam_siaf_aperture_definition.txt [hunk 2]
```diff
@@ -185,6 +212,104 @@ NRCA5_TAGRISMTS_SCI_F322W2 ,      ROI ,   468.0 ,    57.5 ,       32 ,       32
         NRCALL_GRISMR_WFSS , COMPOUND ,    None ,    None ,     None ,     None ,    None ,    None ,                                    NRCALL_FULL ,      grism_wfss
          NRCB5_GRISMC_WFSS ,     SLIT ,    None ,    None ,     None ,     None ,    None ,    None ,                                     NRCB5_FULL ,      grism_wfss
          NRCB5_GRISMR_WFSS ,     SLIT ,    None ,    None ,     None ,     None ,    None ,    None ,                                     NRCB5_FULL ,      grism_wfss
+# Wide Field Slitless Spectroscopy - filter-dependent complete-spectra apertures
```
Comment (shanosborne) [inline pysiaf/source_data/NIRCam/nircam_siaf_aperture_definition.txt]: Remove JWSTSIAF-67 changes from this PR (it's already in #134)

File: pysiaf/source_data/NIRCam/nircam_siaf_aperture_definition.txt (hunk 1 above)
Comment (mgennaro) [reply pysiaf/source_data/NIRCam/nircam_siaf_aperture_definition.txt]: Addressed in incoming commit

**Comment by mgennaro** [comment]
On Jan 16 we encountered problems merging this PR, due to the spcial treatment reserved in the pysiaf code to "master" apertures. What we though was a simple addition of the apertures in JWSTSIAF-164, does in fact not work in a straightforward manner. We thus removed the JWSTSIAF-160 part of the work from this PR and put that in a separate PR #136 . This PR will from now on deal only with JWSTSIAF-164

**Comment by mgennaro** [comment]
The above commit  e6c7df7 solves the issues we previously had when we introduced the new master apertures for dealing with distortions in NIRCam coronography.
Changes:

- introduce new master apertures (4 in total, 2 wedges x 2 channels). These are added in the `siaf_detector_layout` file as well as in the `nircam_siaf_alignment` file
- account for the V2,V3 shifts introduce by the wedges directly in the `nircam_siaf_alignment`. The previous design used a `nircam_siaf_wedge_offset` file and an ad-hoc dependency to achieve the same total offset. Currently the alignment file contains V2,V3 ref positions equal to the corresponding imaging ones, plus the offsets that can be read out of the wedge_offset files. These can in principle be modified in the future
- introduce 4 new distortion files, one per each new master aperture. At this stage these are placeholder files copied from the previous imaging distortion files. they will be replaced with proper solution for the distortion for the coronographic field of view (an optics)
- change the dependency type of ALL the coronographic apertures in module A in the `nircam_siaf_aperture_definitions` file. The dependency is no longer `wedge`, but a simple `default` dependency. The parent apertures for these apertures have been switched from e.g. `NRCA2_FULL` to `NRCA2_FULL_WEDGE_RND` and so on. This by-passes the part of the `generate_nircam` code that was adding the wedge offset to the V2,V3 ref positions of the coronographic apertures.

I have verified in several way that this new design works.
By design the V2,V3 offset I am using is identical to that of the old design, in the wedge file. Also, the distortion coefficients are identical to those used before (cloned from the imaging distortion files). Thus after running generate_nircam, the SIAF.xml file has identical entries as those produced by the old code design, and this is expected. To verify that the code was reading the right inputs, I tweaked the input files in several test runs and verified that the changes propagated downstream into the SIAF.xml file. In particular:

- I tested that by artificially changing the V2,V3 positions in the alignment file, I was obtaining a correspondingly shifted V2,V3 position in the SIAF.xml file
- I tested that by changing the distortion coefficients in the newly created distortion files for coronography, those coefficients end up correctly into the SIAF.xml file

As of now I believe the code changes are complete and everything finally works as expected. The next step will be to incorporate newer distortion coefficients to replace the placeholder ones, and possibly new global offset in the alignment file

**Comment by shanosborne** [comment]
This PR is being closed and replaced by #147; see the replacement PR for a full explanation

**Pull request closed without merging**